### ISSUE TITLE
Remove `ListBoxSelection` in favor of new `next/ListBoxSelection`

### ISSUE THREAD

**Issue by tay1orjones** [carbon-design-system/carbon#12863]
This issue is for refactoring all usages of the legacy `ListBoxSelection` to be updated to use `next/ListBoxSelection`.

A new reimplementation of ListBoxSelection was added in #7777 in `next/ListBoxSelection`. At that time, ComboBox was updated to use the new implementation. Another component and a series of tests were left using the legacy ListBoxSelection and have not been updated.

Attempting to move these out of `next` (https://github.com/carbon-design-system/carbon/issues/12588) caused tests to fail for MultiSelect (https://github.com/carbon-design-system/carbon/pull/12837). Having our tests fail due to this change points to the potential of this being a breaking change. Consumers relying on/testing the tab order of legacy ListBoxSelection via MultiSelect will have one less tab stop with the new implementation. This would break tests in downstream applications without a major version bump.

> ### Initial (now outdated) tasks included
> - [ ] Update and refactor tests in packages/react/src/components/ListBox/__tests__/ListBoxSelection-test.js to use the new ListBoxSelection
> - [ ] Remove legacy ListBoxSelection by moving `next/ListBoxSelection` out of `next` folder to replace the old implementation
> - [ ] Refactor MultiSelect to no longer use legacy ListBoxSelection
> - [ ] Fix any more breaking tests

**Comment by tay1orjones** [comment]
Worth noting that this is not a super pressing issue and only results in slightly inferior accessibility within MultiSelect. I've categorized it as an a11y enhancement.
